src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -80,6 +80,8 @@ def _delete_offer_impl(self, arguments, response_channel):
                 self.offers.remove_offer_from_cache_and_market(self.market, to_delete_offer_id)
             self.state.offered_sell_kWh[self.market.time_slot] -= \
                 self.offers.posted_offer_energy(self.market.id)
+            self.state.energy_to_sell_dict[self.market.time_slot] += \
```

Comment:
This formula adds the total amount of posted offer energy in the energy_to_sell every time a new offer is added. This means that this variable will not be correctly calculated in case there are still pending offers. For instance, if there are 100 posted offers with 1kWh each, when you post a new offer of 1 kWh, the energy_to_sell_dict variable will increase by 101 kWh, which is wrong, because you would like to increase it by one. 
I would recommend to have a look at the `clamp_energy_to_sell` and `clamp_energy_to_buy` methods in the StorageState class, these methods are used for precisely the thing you are trying to achieve in the normal storage strategy.